scripts/partition_manager/partition_manager.rst

```diff
@@ -0,0 +1,408 @@
+
+      spm:
+         placement:
+            after: [mcuboot]
+
+      app:
+         placement:
+            after: [mcuboot]
+
+      foo:
+         span: [mcuboot, app] # The order of spm and app is ambiguous in this case, but since
+                              # this span exists, Partition Manager will try to increase the
+                              # likelihood that mcuboot and app are placed next to each other.
+
+      # Order 1: mcuboot, spm, app
+      # Order 2: mcuboot, app, spm
+      # The algorithm should coerce order 2 to make foo work.
+
+   .. code-block:: yaml
+      :caption: Span example 3 (ambiguous order, cannot work)
+
+      mcuboot:
+         placement:
+
+      spm:
+         placement:
+            after: [mcuboot]
+
+      app:
+         placement:
+            after: [mcuboot]
+
+      foo:
+         span: [mcuboot, app]
+
+      bar:
+         span: [mcuboot, spm]
+
+      # Order 1: mcuboot, spm, app
+      # Order 2: mcuboot, app, spm
+      # foo requires order 2, while bar requires order 1.
+
+inside: list
+   This property is the inverse of ``span``.
+   ``partition_name`` is added to the ``span`` list of the first existing phony partition in the list.
+   This property can be set for image or placeholder partitions.
+
+   .. code-block:: yaml
+      :caption: Example for the inside property
+
+      mcuboot:
+         inside: [b0]
+
+      b0:
+         span: [] # During processing, this span will contain mcuboot.
+
+share_size: list
+   This properties defines the size of the current partition to be the same as the size of the first existing partition in the list.
+   This property can be set for image or placeholder partitions.
+   It cannot be used by phony partitions.
+   The list can contain any kind of partition.
+
+   Note that if the target partition is the ``app`` or a partition that spans over the ``app``, the size is effectively split between them, because the size of the ``app`` is dynamically decided.
+
+.. _pm_yaml_preprocessing:
+
+Configuration file preprocessing
+================================
+
+Each :file:`pm.yml` file is preprocessed to resolve symbols from Kconfig and DTS.
+
+The following example shows what kind of preprocessing is possible:
+
+.. code-block:: yaml
+
+   #include <autoconf.h>
+
+   # 'b0' is the name of the image partition.
+   b0:
+
+     # b0 is placed before the mcuboot partition if the mcuboot partition
+     # exists, otherwise it is stored before the app partition.
+     placement: {before: [mcuboot, app]}
+
+     # The size of the b0 partition is configured in Kconfig.
+     size: CONFIG_BOOTLOADER_PARTITION_SIZE
+
+   # Don't define the provision partition if the SoC is nRF9160, because
+   # the provisioning data is stored in the UICR->OTP data region.
+
+   #ifndef CONFIG_SOC_NRF9160
+
+   # 'provision' is the name of the placeholder partition.
+   provision:
+     # This partition is stored at the very end of flash.
+     placement: last
```

Comment:
We haven't mentioned "last" as valid option anywhere.

Reply:
It should be `end` not `last`. `end` has been mentioned.

Reply:
placement: {before: [end]}